plugins/in_statsd/statsd.c

```diff
@@ -0,0 +1,338 @@
+        if (statsd_process_line(&pck, line) < 0) {
+            flb_error("[in_statsd] failed to process line: '%s'", line);
+        }
+        line = strtok(NULL, "\n");
+    }
+
+    /* Send to output */
+    if (sbuf.size > 0) {
+        flb_input_chunk_append_raw(i_ins, NULL, 0, sbuf.data, sbuf.size);
+    }
+    msgpack_sbuffer_destroy(&sbuf);
+
+    return 0;
+}
+
+static int cb_statsd_init(struct flb_input_instance *i_ins,
+                          struct flb_config *config, void *data)
+{
+    struct flb_statsd *ctx;
+    char *listen;
+    int port;
+
+    ctx = flb_calloc(1, sizeof(struct flb_statsd));
+    if (!ctx) {
+        flb_errno();
+        return -1;
+    }
+
+    ctx->buf = flb_malloc(MAX_PACKET_SIZE);
+    if (!ctx->buf) {
+        flb_errno();
+        flb_free(ctx);
+        return -1;
+    }
+    ctx->i_ins = i_ins;
+
+    /* Listening address */
+    if (i_ins->host.listen) {
+        listen = i_ins->host.listen;
+    }
+    else {
+        listen = DEFAULT_LISTEN;
+    }
+    strncpy(ctx->listen, listen, sizeof(ctx->listen) - 1);
+
+    /* Listening port */
+    if (i_ins->host.port) {
+        port = i_ins->host.port;
+    }
+    else {
+        port = DEFAULT_PORT;
+    }
+    snprintf(ctx->port, sizeof(ctx->port), "%hu", port);
+
+    /* Export plugin context */
+    flb_input_set_context(i_ins, ctx);
+
+    /* Accepts metrics from UDP connections. */
+    ctx->server_fd = flb_net_server_udp(ctx->port, ctx->listen);
+    if (ctx->server_fd == -1) {
+        flb_error("[in_statsd] can't bind to %s:%s", ctx->listen, ctx->port);
+        flb_free(ctx->buf);
+        flb_free(ctx);
+        return -1;
+    }
+
+    /* Set up the UDP connection callback */
+    ctx->coll_fd = flb_input_set_collector_socket(i_ins, cb_statsd_receive,
+                                                  ctx->server_fd, config);
+    if (ctx->coll_fd == -1) {
+        flb_error("[in_statsd] cannot set up connection callback ");
+        flb_socket_close(ctx->server_fd);
+        flb_free(ctx->buf);
+        flb_free(ctx);
+        return -1;
+    }
+
+    flb_info("[in_statsd] start UDP server on %s:%s", ctx->listen, ctx->port);
+
+    return 0;
+}
+
+static void cb_statsd_pause(void *data, struct flb_config *config)
+{
+    struct flb_statsd *ctx = data;
+    flb_input_collector_pause(ctx->coll_fd, ctx->i_ins);
+}
+
+static void cb_statsd_resume(void *data, struct flb_config *config)
+{
+    struct flb_statsd *ctx = data;
+    flb_input_collector_resume(ctx->coll_fd, ctx->i_ins);
+}
+
+static int cb_statsd_exit(void *data, struct flb_config *config)
+{
+    struct flb_statsd *ctx = data;
+
+    flb_input_collector_pause(ctx->coll_fd, ctx->i_ins);
+    flb_socket_close(ctx->server_fd);
+    flb_free(ctx->buf);
+    flb_free(ctx);
+    return 0;
+}
+
+/* Plugin reference */
+struct flb_input_plugin in_statsd_plugin = {
+    .name         = "statsd",
+    .description  = "statsd input plugin",
```

Comment:
put the name in camel case here __StatsD__, that's what the project maintainers use:

https://github.com/statsd/statsd/blob/master/README.md#statsd--

Reply:
> put the name in camel case here StatsD, that's what the project maintainers use:

e13df89 fixed it. Now the patch says "StatsD" whenever appropriate.